smart_contract.sol

```diff
@@ -126,14 +128,30 @@ contract ForkDelta is SafeMath {
     Withdraw(0, msg.sender, amount, tokens[0][msg.sender]);
   }
 
+
   function depositToken(address token, uint amount) public {
     //remember to call Token(address).approve(this, amount) or this contract will not be able to do the transfer on your behalf.
     require(token!=0);
+    depositingTokenFlag = true;
     require(Token(token).transferFrom(msg.sender, this, amount));
+    depositingTokenFlag = false;
```

Comment:
It may be sort of arbitrary, but shouldn't this be changed after the deposit function is complete?

Reply:
It doesn't really matter as long as it is set back to false.

Reply:
Just for cleanliness of code, I think this should be at the end of the function unless you think something could happen and the flag would be set improperly?

Reply:
Sorry, but I do not agree on this. The three lines are one logical unit. I see no benefit from a cleanliness point of view.

Reply:
Fair enough.